packages/roosterjs-content-model-api/lib/modelApi/table/insertTableColumn.ts

```diff
@@ -23,11 +23,18 @@ export function insertTableColumn(
             table.rows.forEach(row => {
                 const cell = row.cells[insertLeft ? sel.firstColumn : sel.lastColumn];
 
+                const cellFormat =
+                    cell.blocks.length > 0 &&
+                    cell.blocks[0].blockType == 'Paragraph' &&
+                    cell.blocks[0].segments.length > 0
+                        ? { ...cell.format, ...cell.blocks[0].segments[0].format }
```

Comment:
There is code to help only copy segment format, https://github.com/microsoft/roosterjs/blob/84ac4284cde532cb3952a3af02e7a2afcc2a5ced/packages/roosterjs-content-model-core/lib/corePlugin/cache/domIndexerImpl.ts#L470

You can extract it to be a public function so call it from both places. This is to avoid copy additional format, for example, image.